paperqa/agents/search.py

```diff
@@ -326,6 +344,14 @@ async def _add_document() -> None:
             )
             raise
 
+    async def commit(self) -> None:
+        """Commit all pending changes to the index."""
```

Comment:
```suggestion
    async def checkpoint(self) -> None:
        """Commit all pending changes, merge threads, and reset."""
```

This isn't just a `commit` any more, let's call it `checkpoint`